src/components/Settings/preferences/accordions/LanguageAccordion/LanguageAccordion.tsx

```diff
@@ -55,6 +57,7 @@ export const LanguageAccordion: React.FC<LanguageAccordionProps> = ({
         enqueueSnackbar(t('Saved successfully.'), {
           variant: 'success',
         });
+        setLocaleStorageLanguage(attributes.locale || '');
```

Comment:
My hope is that this wouldn't be necessary. Changing the preference load the new preference from the server, which will update the locale in the Apollo cache, which will cause `user.preferences.locale` to update in `UseApiConstants`.